src/main/scala/spray/json/package.scala

```diff
@@ -32,7 +32,7 @@ package object json {
 
 package json {
 
-  class DeserializationException(msg: String, cause: Throwable = null) extends RuntimeException(msg, cause)
+  case class DeserializationException(msg: String, cause: Throwable = null, fieldName: Option[String] = None) extends RuntimeException(msg, cause)
```

Comment:
How about we turn `fieldName` into `fieldNames: List[String]` so we can gather the *path* to the error field rather than just the inner most member name?